src/components/Microsoft.DotNet.UpgradeAssistant/Checks/BypassableReadinessCheck.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Checks
+{
+    public abstract class BypassableReadinessCheck : IUpgradeReadyCheck
+    {
+        private readonly ILogger _logger;
+        private readonly UpgradeOptions _upgradeOptions;
+
+        protected BypassableReadinessCheck(ILogger logger, UpgradeOptions upgradeOptions)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _upgradeOptions = upgradeOptions ?? throw new ArgumentNullException(nameof(upgradeOptions));
+        }
+
+        public abstract string Id { get; }
+
+        public abstract bool IsBypassable { get; }
+
+        public abstract string TechnologyDetected { get; }
+
+        public abstract string SupportLink { get; }
```

Comment:
Should this be a Url?

Reply:
I'll add comment to clarify. The intent is to provide guidance to address the technology detected. Ideally, there would be one landing page for this discussion but the string gives us flexibility to provide an explanation and 2 links as needed.

E.g.
- https://docs.microsoft.com/en-us/dotnet/architecture/grpc-for-wcf-developers/migrate-wcf-to-grpc
- https://docs.microsoft.com/en-us/dotnet/core/porting/net-framework-tech-unavailable?toc=/dotnet/fundamentals/toc.json&bc=/dotnet/breadcrumb/toc.json#workflow-foundation-and-wcf

Also, note to self, I had meant to share both links in the code.